HOWTO-make-a-release.md

```diff
@@ -0,0 +1,348 @@
+# HOW TO MAKE A RELEASE
+
+This file documents how to make an OpenSSL release.  Please fix any
+errors you find while doing, or just after, your next release!
+
+Releases are done by one person, with a second person acting as the
+reviewer and additional tester.
+
+# Table of contents
+
+-   [Prerequisites](#prerequisites)
+    -   [Check your access](#check-your-access)
+    -   [PGP / GnuPG key](#pgp-gnupg-key)
+    -   [Repositories](#repositories)
+    -   [A method for reviewing](#a-way-to-reviewing)
+-   [Pre-publishing tasks](#pre-publishing-tasks) [once for each version being released]
+    -   [Freeze the source repository](#freeze-the-source-repository) [the day before release]
+    -   [Prepare your repository checkouts](#prepare-your-repository-checkouts)
+    -   [Make sure that the source is up to date](#make-sure-that-the-source-is-up-to-date)
+    -   [Generate the tarball and announcement text](#generating-the-tarball-and-announcement-text) [do not push]
+        -   [OpenSSL 3.0 and on](#openssl-3.0-and-on)
+        -   [OpenSSL before 3.0](#openssl-before-3.0)
+    -   [Update the website locally](#update-the-website-locally) [do not push]
+-   [Publish the release](#publish-the-release)
+-   [Post-publishing tasks](#post-publishing-tasks)
+    -   [Check the website](#check-the-website)
+    -   [Send the announcement mail](#send-the-announcement-mail)
+    -   [Unfreeze the source repository](#unfreeze-the-source-repository)
+    -   [Security fixes](#security-fixes)
+    -   [Keep in touch](#keep-in-touch)
+
+
+# Prerequisites
+
+## Check your access
+
+To perform a release, you must have appropriate access to OpenSSL's
+development host, dev.openssl.org.  To test this, try to log in with
+ssh:
+
+    ssh dev.openssl.org
+
+You must also check that you can perform tasks as the user 'openssl'
+on dev.openssl.org.  When you have successfully logged in, test your
+access to that user with sudo:
+
+    sudo -u openssl id
+
+## PGP / GnuPG key
+
+You must have a PGP / GnuPG key, and its fingerprint should be present
+in the file `doc/fingerprints.txt` in the OpenSSL source.
```

Comment:
Really, it should have been present in fingerprints.txt in the immediately prior release - so that we have a chain of trust from one release to the next. This does imply some forward planning about who we want to do releases.

Reply:
Agreed...  do you want me to alter the text somehow?

Reply:
How about:

You must have a PGP / GnuPG key, and its fingerprint should be present in the file `doc/fingerprints.txt` in the source of the immediately prior OpenSSL release.

Reply:
Works with me!